### ISSUE TITLE
Support zstd for "log-driver": "local"

### ISSUE THREAD

**Issue by HectorB-2020** [moby/moby#43961]
### Description

Currently `log-driver` `local` supports only one parameter related to compression: `true` or `false`.
```
compress | Toggle compression of rotated log files. Enabled by default.
```
And log files are rotated compressed only as `gzip`, no other options.

I propose adding support for other promising algorithm - Zstandard, also known as `std`.

**Comment by thaJeztah** [comment]
Is there a specific reason for this request? Are you looking at reducing size consumed on disk, or at performance improvements?

I'm personally not against this _but_ if someone wants to try implementing this, we should have benchmarks for comparison before we merge. I recall that (at least some time ago) `zstd` could provide substantial improvements when _decompressing_ (so when reading back the logs), but at the cost of being slower (and more resource intensive) during _compression_.

If that's still the case, we should look at how much the additional overhead is, as writing the logs (compressing) would happen for all containers using the logging driver (irregardless if those docs are read by someone), and reading _only_ if logs are read back.

**Comment by cpuguy83** [comment]
We're also writing some extra data to the gzip headers to optimize searches.
I think to support zstd we'd really need a strong case for it.

**Comment by thaJeztah** [comment]
Good point: didn't know (or forgot about) the extra data in the headers. Yes, so curious about the use case, and definitely would like to see a comparison to see if there's enough benefits to make it worth changing.

**Comment by HectorB-2020** [comment]
> We're also writing some extra data to the gzip headers to optimize searches

Good to know, thanks! I wasn't aware of this.

In fact you are right: the rationale behind that request was both performance and space saving. Colleagues of mine ran research of zstd for other logs, found results very promising and we've made company-wide decision to switch all our log rotation to zstd.

**Comment by thaJeztah** [comment]
I think (from the discussion above); it's not a hard "reject", but indeed if someone is interested in looking if it brings significant improvements in this specific case.

**Comment by gaby** [comment]
Now days parquet, arrow, vector databases, etc all use Zstd. The compression speed and reduction in size of logs, text, and JSON is quite significant.
